scripts/target.js

```diff
@@ -0,0 +1,281 @@
+const SESSION_COOKIE_NAME = 'sessionId';
+const SESSION_COOKIE_EXPIRATION_DAYS = 7;
+
+/**
+ * Escape a selector.
+ * @param selector
+ * @returns {string}
+ */
+function escapeSelector(selector) {
+  return selector.replaceAll(/#(\d)/g, '#\\3$1 ');
+}
+
+/**
+ * Generate a UUID.
+ * @returns {string}
+ */
+function uuid() {
+  return `sess-${Math.random()
+    .toString(36)
+    .substring(2, 9)}-${Date.now()
+    .toString(36)}`;
+}
+
+/**
+ * Get a cookie by name.
+ * @param name
+ * @returns {string}
+ */
+function getCookie(name) {
+  const cookies = document.cookie.split(';');
+  return cookies.find((cookie) => {
+    const [key, value] = cookie.split('=');
+    return key.trim() === name && value;
+  });
+}
+
+/**
+ * Set a cookie.
+ * @param name
+ * @param value
+ * @param days
+ */
+function setCookie(name, value, days) {
+  document.cookie = `${name}=${value}; max-age=${days * 24 * 60 * 60}; path=/`;
+}
+
+/**
+ * Get the session ID.
+ * @returns {string}
+ */
+function getSessionId() {
+  const existingSessionId = getCookie(SESSION_COOKIE_NAME);
+  if (existingSessionId) {
+    return existingSessionId;
+  }
+  const newSessionId = uuid();
+  setCookie(SESSION_COOKIE_NAME, newSessionId, SESSION_COOKIE_EXPIRATION_DAYS);
+  return newSessionId;
+}
+
+/**
+ * Get all offers from a response.
+ * @param data
+ * @returns {*[]}
+ */
+function getApplicableOffers(data) {
+  const offers = [];
+  const options = data.execute?.pageLoad?.options ?? [];
+  console.debug(`received ${options.length} options`); // eslint-disable-line no-console
+  options.forEach((option) => {
+    if (option.type === 'actions') {
+      option.content.forEach((content) => {
+        console.debug('processing content', content); // eslint-disable-line no-console
+        if (['setHtml', 'insertAfter', 'insertBefore'].includes(content.type)) {
+          offers.push(content);
+        }
+      });
+    }
+  });
+  return offers;
+}
+
+/**
+ * Fetch offers for a client and a host.
+ * @param client
+ * @param sessionId
+ * @param useProxy Whether to use the proxy.
+ * @returns {Promise<any>}
+ */
+async function fetchOffers(client, sessionId, useProxy) {
+  const url = `${window.location.protocol}//${window.location.host}`;
+
+  console.debug(`Loading offers for client ${client} and url ${url}`); // eslint-disable-line no-console
+
+  const payload = {
+    context: {
+      channel: 'web',
+      address: {
+        url,
+      },
+    },
+    execute: {
+      pageLoad: {},
+    },
+  };
+
+  const options = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'cache-control': 'no-cache',
+    },
+    body: JSON.stringify(payload),
+  };
+
+  const host = useProxy ? '/' : `https://${client}.tt.omtrdc.net/`;
+  console.debug(`Using target host: ${host}`); // eslint-disable-line no-console
+  const response = await fetch(`${host}rest/v1/delivery?client=${client}&sessionId=${sessionId}`, options);
```

Comment:
You could use http headers instead of query parameters so we don't leak those sensitive info in the logs potentially